packages/babel-types/src/definitions/core.js

```diff
@@ -303,8 +303,8 @@ defineType("Identifier", {
   fields: {
     name: {
       validate(node, key, val) {
-        if (!t.isValidIdentifier(val)) {
-          // todo
+        if (!t.isSpecIdentifierName(val)) {
+          throw new TypeError(`Property "${key}" of ${node.type} failed validation with value: ${JSON.stringify(val)}. Expected true for t.isSpecIdentifierName().`);
```

Comment:
@thejameskyle I was wondering that originally, but **sebmck** [said it's abandoned](https://babeljs.slack.com/archives/development/p1450104369000551).